crates/bevy_derive/src/modules.rs

```diff
@@ -1,65 +1,32 @@
-use find_crate::Manifest;
-use proc_macro::TokenStream;
-use syn::{Attribute, Path};
+use bevy_macro_utils::{get_module_path, get_path};
+use syn::Attribute;
 
-#[derive(Debug)]
 pub struct Modules {
-    pub bevy_render: String,
-    pub bevy_asset: String,
-    pub bevy_core: String,
-    pub bevy_utils: String,
-    pub bevy_app: String,
-}
-
-impl Modules {
-    pub fn meta(name: &str) -> Modules {
-        Modules {
-            bevy_asset: format!("{}::asset", name),
-            bevy_render: format!("{}::render", name),
-            bevy_core: format!("{}::core", name),
-            bevy_utils: format!("{}::utils", name),
-            bevy_app: format!("{}::app", name),
-        }
-    }
-
-    pub fn external() -> Modules {
-        Modules {
-            bevy_asset: "bevy_asset".to_string(),
-            bevy_render: "bevy_render".to_string(),
-            bevy_core: "bevy_core".to_string(),
-            bevy_utils: "bevy_utils".to_string(),
-            bevy_app: "bevy_app".to_string(),
-        }
-    }
-}
-
-fn get_meta() -> Option<Modules> {
-    let manifest = Manifest::new().unwrap();
-    if let Some(package) = manifest.find(|name| name == "bevy") {
-        Some(Modules::meta(&package.name))
-    } else if let Some(package) = manifest.find(|name| name == "bevy_internal") {
-        Some(Modules::meta(&package.name))
-    } else {
-        None
-    }
+    pub bevy_app: syn::Path,
+    pub bevy_asset: syn::Path,
+    pub bevy_core: syn::Path,
+    pub bevy_render: syn::Path,
+    pub bevy_utils: syn::Path,
 }
 
 const AS_CRATE_ATTRIBUTE_NAME: &str = "as_crate";
 
 pub fn get_modules(attributes: &[Attribute]) -> Modules {
```

Comment:
This refactor forces a "re-parse" of the entire manifest toml 5 times for every call to get_modules. We should cut that down to 1.

Reply:
Done and rebased to resolve a conflict.